sympy/core/add.py

```diff
@@ -703,7 +703,7 @@ def _eval_is_zero(self):
                 return
         if z == len(self.args):
             return True
-        if len(nz) == 0 or len(nz) == len(self.args):
+        if not nz or len(nz) == len(self.args):
```

Comment:
Why not `len(nz) in [0, len(self.args)]` for this one?

Reply:
I think I had that, but then made a guess that not nz should be faster to evaluate (as checking len() == 0 is often reported as a bad idea by linters). But it is indeed a guessing game.